numpy/core/src/multiarray/ctors.c

```diff
@@ -513,29 +513,28 @@ setArrayFromSequence(PyArrayObject *a, PyObject *s,
         goto fail;
     }
 
-    slen = PySequence_Length(s);
-    if (slen < 0) {
+    /* Try __array__ before using s as a sequence */
+    PyObject *tmp = _array_from_array_like(
+            s, /*dtype*/ NULL, /*writeable*/ 0, /*context*/ NULL);
```

Comment:
nit: Tempting to change this to match the other call exactly, either as `_array_from_array_like(obj, NULL, NPY_FALSE, NULL);` here or this way there.

Reply:
Sure, presonally I now use an IDE which actually adds hints for the names here, giving it double...

Reply:
I've seen a former coworker use that before, it looked pretty cool